plugins/lang/js/js_common.cc

```diff
@@ -75,17 +233,39 @@ init(bool forThisThreadOnly)
 }
 
 V8Init::
-V8Init()
+V8Init(MldbServer * server)
 {
+    static std::atomic<bool> alreadyDone(false);
+    if (alreadyDone)
+        return;
+
+    static std::mutex mutex;
+    std::unique_lock<std::mutex> guard(mutex);
+
+    if (alreadyDone)
+        return;
+
     const char * v8_argv[] = {
         "--lazy", "false"
         ,"--always_opt", "true"
-        ,"--always_full_compiler", "true"
-        //,"--trace-opt", "true"
     };
     int v8_argc = sizeof(v8_argv) / sizeof(const char *);
 
     v8::V8::SetFlagsFromCommandLine(&v8_argc, (char **)&v8_argv, false);
+
+    // TODO: linuxisms...
+    char exePath[PATH_MAX];
+    ssize_t pathLen = readlink("/proc/self/exe", exePath, PATH_MAX);
+    if (pathLen == -1)
+        throw HttpReturnException(400, "Couldn't determine v8 library path");
```

Comment:
The message is misleading.  We should say that we cannot find our own path.  And I feel this should be a 500.  Not sure how a user could fix that.